rust/src/protocol_types/governance/voter.rs

```diff
@@ -0,0 +1,112 @@
+use crate::*;
+
+#[derive(
+    Clone,
+    Debug,
+    Hash,
+    Eq,
+    Ord,
+    PartialEq,
+    PartialOrd,
+    serde::Serialize,
+    serde::Deserialize,
+    JsonSchema,
+)]
+pub(crate) enum VoterEnum {
+    ConstitutionalCommitteeHotKey(Credential),
+    DRep(Credential),
+    StakingPool(Ed25519KeyHash),
+}
+
+#[wasm_bindgen]
+#[derive(Clone, Debug, Hash, Eq, Ord, PartialEq, PartialOrd)]
+pub enum VoterKind {
+    ConstitutionalCommitteeHotKeyHash,
+    ConstitutionalCommitteeHotScriptHash,
+    DRepKeyHash,
+    DRepScriptHash,
+    StakingPoolKeyHash,
+}
+
+#[derive(
+    Clone,
+    Debug,
+    Hash,
+    Eq,
+    Ord,
+    PartialEq,
+    PartialOrd,
+    serde::Serialize,
+    serde::Deserialize,
+    JsonSchema,
+)]
+#[wasm_bindgen]
+pub struct Voter(pub(crate) VoterEnum);
+
+impl_to_from!(Voter);
+
+#[wasm_bindgen]
+impl Voter {
+    pub fn new_constitutional_committee_hot_key(cred: &Credential) -> Self {
+        Self(VoterEnum::ConstitutionalCommitteeHotKey(cred.clone()))
+    }
+
+    pub fn new_drep(cred: &Credential) -> Self {
+        Self(VoterEnum::DRep(cred.clone()))
+    }
+
+    pub fn new_staking_pool(key_hash: &Ed25519KeyHash) -> Self {
+        Self(VoterEnum::StakingPool(key_hash.clone()))
+    }
+
+    pub fn kind(&self) -> VoterKind {
+        match &self.0 {
+            VoterEnum::ConstitutionalCommitteeHotKey(cred) => match cred.kind() {
+                CredKind::Key => VoterKind::ConstitutionalCommitteeHotKeyHash,
+                CredKind::Script => VoterKind::ConstitutionalCommitteeHotScriptHash,
+            },
+            VoterEnum::DRep(cred) => match cred.kind() {
+                CredKind::Key => VoterKind::DRepKeyHash,
+                CredKind::Script => VoterKind::DRepScriptHash,
+            },
+            VoterEnum::StakingPool(_) => VoterKind::StakingPoolKeyHash,
+        }
+    }
+
+    pub fn to_constitutional_committee_hot_cred(&self) -> Option<Credential> {
```

Comment:
Thanks for point it out, we will consider renaming of it

Reply:
We decided to name it `to_constitutional_committee_hot_credential` since it reflects it is a credential not just a key and to have naming consistency with CC auth and resign certs.